src/CoreBundle/Service/RequestInfoService.php

```diff
@@ -129,6 +132,27 @@ public function isCmsTemplateEngineEditMode()
         return $this->isCmsTemplateEngineEditModeCache;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isPreviewMode()
+    {
+        if (null !== $this->isPreviewModeCache) {
+            return $this->isPreviewModeCache;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+        if (null === $request) {
+            return false;
+        }
+
+        // todo: `__rpeviewmode` should be the only way to enable this. Refactor all places where the preview attribute is set as `preview` instead of `__previewmode'
```

Comment:
```suggestion
        // todo: `__previewmode` should be the only way to enable this. Refactor all places where the preview attribute is set as `preview` instead of `__previewmode'
```

Reply:
fixed